experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -83,7 +83,10 @@ LogicalResult HandshakeSpeculationPass::placeUnits(Value ctrlSignal) {
     inheritBB(dstOp, newOp);
 
     // Connect the new Operation to dstOp
-    srcOpResult.replaceAllUsesExcept(newOp.getResult(), newOp);
+    // Note: srcOpResult.replaceAllUsesExcept cannot be used here
+    // because the uses of srcOpResult may include a newly created
+    // operand for the speculator enable signal.
+    operand->set(newOp.getResult());
```

Comment:
I guess replaceAllUsesExcept may have a bug in other places too (like placeSpeculator method). Probably it should also be replaced to `operand->set`

Reply:
this comment is not clear enough- I think you need to be extremely verbose with an exact example of what the problem is

Reply:
This figure illustrates the example. Should I describe this figure in text here (using chatgpt)?
https://github.com/EPFL-LAP/dynamatic/pull/168#issuecomment-2422817334

Reply:
yeah your explanation in the repo is great, it just also needs to be in the code

Reply:
I tried to provide an explanation so please take a look!